src/pkg/packager/prepare.go

```diff
@@ -200,9 +215,15 @@ func (p *Packager) FindImages() (imgMap map[string][]string, err error) {
 		defer spinner.Stop()
 
 		for _, resource := range resources {
-			if matchedImages, maybeImages, err = p.processUnstructuredImages(resource, matchedImages, maybeImages); err != nil {
+			var whyImageResourcesPartial []*unstructured.Unstructured
+			if matchedImages, maybeImages, whyImageResourcesPartial, err = p.processUnstructuredImages(resource, matchedImages, maybeImages, whyImage); err != nil {
 				message.WarnErrf(err, "Problem processing K8s resource %s", resource.GetName())
 			}
+			updatedResources := append(whyImageResources[component.Name].Resources, whyImageResourcesPartial...)
+			whyImageResources[component.Name] = manifestResourceRelation{
+				Manifests: component.Manifests,
+				Resources: updatedResources,
+			}
```

Comment:
Doing this here will match up references for Zarf `manifests` but not Zarf `charts` (which are also templated in find-images).

Running `zarf dev find-images examples/helm-charts/ --why curlimages/curl:7.69.0` results in many resources being assigned to the same empty string manifest when they really belong to different charts in the component:

![image](https://github.com/defenseunicorns/zarf/assets/3977569/7f75edec-e122-4105-b662-ef9fa4ec30e7)

Reply:
Could have made that clearer in the ACs but the above should result in something like (note the swap from `manifest` to `chart`):

```
component: demo-helm-charts
chart: podinfo-oci
resources: 

apiVersion: v1
kind: Pod
metadata:
  annotations:
    appmesh.k8s.aws/sidecarInjectorWebhook: disabled
    helm.sh/hook: test-success
    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
    linkerd.io/inject: disabled
    sidecar.istio.io/inject: "false"
  labels:
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: podinfo-oci
    app.kubernetes.io/version: 6.4.0
    helm.sh/chart: podinfo-6.4.0
  name: podinfo-oci-service-test-xaab9
spec:
  containers:
  - command:
    - sh
    - -c
    - |
      curl -s ${PODINFO_SVC}/api/info | grep version
    env:
    - name: PODINFO_SVC
      value: podinfo-oci.podinfo-from-oci:9898
    image: curlimages/curl:7.69.0
    name: curl
  restartPolicy: Never

component: demo-helm-charts
chart: podinfo-git
resources: 

apiVersion: v1
kind: Pod
metadata:
  annotations:
    appmesh.k8s.aws/sidecarInjectorWebhook: disabled
    helm.sh/hook: test-success
    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
    linkerd.io/inject: disabled
    sidecar.istio.io/inject: "false"
  labels:
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: podinfo-git
    app.kubernetes.io/version: 6.4.0
    helm.sh/chart: podinfo-6.4.0
  name: podinfo-git-service-test-ioazw
spec:
  containers:
  - command:
    - sh
    - -c
    - |
      curl -s ${PODINFO_SVC}/api/info | grep version
    env:
    - name: PODINFO_SVC
      value: podinfo-git.podinfo-from-git:9898
    image: curlimages/curl:7.69.0
    name: curl
  restartPolicy: Never

component: demo-helm-charts
chart: podinfo-repo
resources: 

apiVersion: v1
kind: Pod
metadata:
  annotations:
    appmesh.k8s.aws/sidecarInjectorWebhook: disabled
    helm.sh/hook: test-success
    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
    linkerd.io/inject: disabled
    sidecar.istio.io/inject: "false"
  labels:
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: cool-release-name-podinfo
    app.kubernetes.io/version: 6.4.0
    helm.sh/chart: podinfo-6.4.0
  name: cool-release-name-podinfo-service-test-h0q8g
spec:
  containers:
  - command:
    - sh
    - -c
    - |
      curl -s ${PODINFO_SVC}/api/info | grep version
    env:
    - name: PODINFO_SVC
      value: cool-release-name-podinfo.podinfo-from-repo:9898
    image: curlimages/curl:7.69.0
    name: curl
  restartPolicy: Never

```